ModularTegustation/tegu_items/lc13/refinery/sales.dm

```diff
@@ -0,0 +1,104 @@
+/obj/structure/pe_sales
+	name = "PE Power Generation"
+	desc = "A machine used to send refined PE to the respective company"
+	icon = 'ModularTegustation/Teguicons/refiner.dmi'
+	icon_state = "machinelcb"
+	anchored = TRUE
+	density = FALSE
+	resistance_flags = INDESTRUCTIBLE
+	var/crate = /obj/structure/lootcrate
+	var/ahn_amount = 400 	//gives you a random amount of ahn between this number and 1/4th this number
+	var/power_timer = 120 	//How long does the box last for? You get 1 point every second
+	var/crate_timer = 180	//How much time until a crate?
+	var/crates_per_box		//Just used to calculate examine text
+
+	var/generating
+	var/icon_full = "machinelcb_full"
+
+/obj/structure/pe_sales/Initialize()
+	..()
+	crates_per_box = crate_timer/power_timer
+
+/obj/structure/pe_sales/examine(mob/user)
+	. = ..()
+	. += "The number of crates required to recieve payment is [crates_per_box]"
+
+/obj/structure/pe_sales/attackby(obj/item/I, mob/living/user, params)
+	if(istype(I, /obj/item/rawpe) && !generating)
+		to_chat(user, "<span class='notice'>You need to refine this first!.</span>")
+
+	else if(istype(I, /obj/item/refinedpe) && !generating)
+		generating = TRUE
+		to_chat(user, "<span class='notice'>You load PE into the machine.</span>")
+		qdel(I)
+		counter()
+		add_overlay("full")
+	else if(generating)
+		to_chat(user, "<span class='notice'>This is already sending power!</span>")
+
+/obj/structure/pe_sales/proc/counter()
+	power_timer--
+	crate_timer--
+
+	//Box is done
+	if(power_timer <= 0)
+		power_timer = initial(power_timer)
+		generating = FALSE
+		visible_message("<span class='notice'>Payment has arrived from [src]</span>")
+		cut_overlays()
+		var/obj/item/holochip/C = new (get_turf(src))
+		C.credits = rand(ahn_amount/4,ahn_amount)
+		if(SSlobotomy_corp.box_goal <=1500)
+			SSlobotomy_corp.box_goal-=50	//Quota goes down, you're doing the job for them.
+		else
+			SSlobotomy_corp.box_goal=1500
```

Comment:
I think a better solution to this may just be adding to the current value, you know?

Reply:
That would allow for infinite PE so long as you're generating boxes